### DISCUSSION TITLE
Question | SSR

### DISCUSSION THREAD

**Discussion by samuelcastro** [UnlyEd/next-right-now#170] [General] [answered]
I'm wondering why you're not returning `customer` here: https://github.com/UnlyEd/next-right-now/blob/v2-mst-aptd-gcms-lcz-sty/src/utils/nextjs/SSR.ts#L129 and https://github.com/UnlyEd/next-right-now/blob/v2-mst-aptd-gcms-lcz-sty/src/utils/nextjs/SSR.ts#L29. 

It'd be helpful to keep it as part of `getCommonServerSideProps` to avoid duplication, I might be wrong but I don't see limitations on SSR to why we can't leave it in the `getCommonServerSideProps` function.

**Comment by Vadorequest** [comment, marked as answer]
Because it's a matter of optimization.

With SSG, we don't care much about optimization, since it won't slow down real users, but with SSR it will, so we need to be more careful.

Because each GQL request depends on what page is being displayed, we don't preload the `customer` because we want to load all the data that page needs at once, and we can't know that from `getExamplesCommonServerSideProps`.

Instead, we do it on the page itself, and that way we only run one GQL query.

**Reply by samuelcastro** [reply]
Ok that make sense.
